engine/gfx/dynamic_pals.asm

```diff
@@ -147,21 +147,19 @@ MarkUsedPal:
 	push bc
 
 	; Pal is not already loaded, find a empty pal slot
-	lb bc, 0, 8
-	ld hl, wUsedObjectPals
-	ld a, 1
-	ld d, a
-.search_again
-	ld a, d
-	and [hl]
-	jr z, .found_empty
-	ld a, d
-	rla
-	ld d, a
+	ld a, [wUsedObjectPals]
+	cp %11111111
```

Comment:
You can do `inc a :: jr z, .none_available :: dec a` ([wiki](https://github.com/pret/pokecrystal/wiki/Optimizing-assembly-code#compare-a-to-255)). Will think about the loop later; should be possible to use the carry flag instead of `bit 0` checks.